protocol/synthetix/subgraph/mainnet/handleCollateralConfigured.ts

```diff
@@ -17,5 +20,18 @@ export function handleCollateralConfigured(event: CollateralConfigured): void {
   collateralType.min_delegation = event.params.config.minDelegationD18.toBigDecimal();
   collateralType.updated_at = event.block.timestamp;
   collateralType.updated_at_block = event.block.number;
+
+  const erc20Contract = ERC20.bind(event.params.collateralType);
+
+  if (erc20Contract) {
+    collateralType.name = erc20Contract.name();
```

Comment:
Normally I would say we should avoid calling contracts from event handlers as it is incredibly slow comparing to taking events values.
In this case it is less of an issue as this particular event happens very infrequently.

I am still convinced we should not have this info in subgraph though as this info can be easily retrieved from the UI and we should avoid using subgraphs for anything that is possible (and feasible) to do with contracts directly from the UI.

Still not convinced this is necessary.

Reply:
The idea is simple: only use event values in graph handlers. And add contrast calls only if there is absolutely no other way accessing data.
We already have v2 subgraph that did not consider any restrictions.